dot/node.go

```diff
@@ -41,9 +48,84 @@ type Node struct {
 	metricsServer *metrics.Server
 }
 
+//go:generate mockgen -source=node.go -destination=mock_node_test.go -package=$GOPACKAGE
+
+type nodeBuilderIface interface {
+	nodeInitialised(string) error
+	initNode(config *Config) error
+	createStateService(config *Config) (*state.Service, error)
+	createNetworkService(cfg *Config, stateSrvc *state.Service, telemetryMailer telemetry.Client) (*network.Service,
+		error)
+	createRuntimeStorage(st *state.Service) (*runtime.NodeStorage, error)
+	loadRuntime(cfg *Config, ns *runtime.NodeStorage, stateSrvc *state.Service, ks *keystore.GlobalKeystore,
+		net *network.Service) error
+	createBlockVerifier(st *state.Service) (*babe.VerificationManager, error)
+	createDigestHandler(lvl log.Level, st *state.Service) (*digest.Handler, error)
+	createCoreService(cfg *Config, ks *keystore.GlobalKeystore, st *state.Service, net *network.Service,
+		dh *digest.Handler) (*core.Service, error)
+	createGRANDPAService(cfg *Config, st *state.Service, dh *digest.Handler, ks keystore.Keystore,
+		net *network.Service, telemetryMailer telemetry.Client) (*grandpa.Service, error)
+	newSyncService(cfg *Config, st *state.Service, fg dotsync.FinalityGadget, verifier *babe.VerificationManager,
+		cs *core.Service, net *network.Service, telemetryMailer telemetry.Client) (*dotsync.Service, error)
+	createBABEService(cfg *Config, st *state.Service, ks keystore.Keystore, cs *core.Service,
+		telemetryMailer telemetry.Client) (*babe.Service, error)
+	createSystemService(cfg *types.SystemInfo, stateSrvc *state.Service) (*system.Service, error)
+	createRPCService(params rpcServiceSettings) (*rpc.HTTPServer, error)
+}
+
+type nodeBuilder struct{}
```

Comment:
Updated, what does this do?

Reply:
Just compile check the nodeBuilder implements the `nodeBuilderIface`. That should be compiled check elsewhere as well, but it's a good habit to have to easily find why something is going wrong.

It's also rather critical if you have a Go API/library without an actual program using the implementation, and if you have 2 implementations for one interface. In this case it's critical to make sure they both implement the common interface since there is no other compilation check otherwise.